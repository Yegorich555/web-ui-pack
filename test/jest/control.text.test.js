import { WUPTextControl } from "web-ui-pack";
import { initTestBaseControl, testBaseControl } from "./baseControlTest";

/** @type WUPTextControl */
let el;
initTestBaseControl({ type: WUPTextControl, htmlTag: "wup-text", onInit: (e) => (el = e) });

describe("control.text", () => {
  testBaseControl({
    initValues: [
      { attrValue: "123", value: "123" },
      { attrValue: "some txt", value: "some txt" },
      { attrValue: "34", value: "34" },
    ],
  });

  test("selectAll by clear/focus", () => {
    // todo split test for $option.selectOnFocus
    // todo test btnClear
    const initV = "some text";
    el.$initValue = initV;
    expect(el.$refInput.value).toBe(initV);
    el.focus();
    expect(el.$refInput.selectionStart).toBe(0);
    expect(el.$refInput.selectionEnd).toBe(initV.length);

    el.$refInput.selectionEnd = 0;
    expect(el.$refInput.selectionEnd).toBe(0);

    el.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
    expect(el.$value).toBe(undefined);
    el.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
    expect(el.$value).toBe(initV);

    expect(el.$refInput.selectionStart).toBe(0);
    expect(el.$refInput.selectionEnd).toBe(initV.length);
  });
});
