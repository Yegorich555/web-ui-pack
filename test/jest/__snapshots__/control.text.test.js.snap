// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control.text $show/$hide error 1`] = `
<wup-text>
  <label
    for="txt1"
  >
    <span>
      <input
        autocomplete="off"
        id="txt1"
        placeholder=" "
        type="text"
      />
      <strong />
    </span>
    <button
      aria-hidden="true"
      clear=""
      tabindex="-1"
    />
  </label>
</wup-text>
`;

exports[`control.text $show/$hide error 2`] = `
<wup-text
  invalid=""
>
  <label
    for="txt1"
  >
    <span>
      <input
        aria-describedby="txt2"
        autocomplete="off"
        id="txt1"
        placeholder=" "
        type="text"
      />
      <strong />
    </span>
    <button
      aria-hidden="true"
      clear=""
      tabindex="-1"
    />
  </label>
  <wup-popup
    aria-atomic="true"
    aria-live="off"
    error=""
    id="txt2"
    role="alert"
  >
    <span
      class="wup-hidden"
    >
      Error for :
    </span>
    <span>
      Some custom message here
    </span>
  </wup-popup>
</wup-text>
`;

exports[`control.text $show/$hide error 3`] = `
<wup-text>
  <label
    for="txt1"
  >
    <span>
      <input
        autocomplete="off"
        id="txt1"
        placeholder=" "
        type="text"
      />
      <strong />
    </span>
    <button
      aria-hidden="true"
      clear=""
      tabindex="-1"
    />
  </label>
</wup-text>
`;
